src/System.Management.Automation/engine/Utils.cs

```diff
@@ -895,113 +895,61 @@ internal static bool IsAdministrator()
 #endif
         }
 
-        internal static bool NativeItemExists(string path)
+        internal static bool ItemExists(string path)
         {
-            bool unusedIsDirectory;
-            Exception unusedException;
+            bool result = false;
+            try
+            {
+                result = ItemExists(path, out bool unused);
+            }
+            catch
+            {
+            }
 
-            return NativeItemExists(path, out unusedIsDirectory, out unusedException);
+            return result;
         }
 
-        // This is done through P/Invoke since File.Exists and Directory.Exists pay 13% performance degradation
-        // through the CAS checks, and are terribly slow for network paths.
-        internal static bool NativeItemExists(string path, out bool isDirectory, out Exception exception)
+        internal static bool ItemExists(string path, out bool isDirectory)
         {
-            exception = null;
+            isDirectory = false;
 
             if (String.IsNullOrEmpty(path))
             {
-                isDirectory = false;
                 return false;
             }
-#if UNIX
-            isDirectory = Platform.NonWindowsIsDirectory(path);
-            return Platform.NonWindowsIsFile(path);
-#else
-
+#if !UNIX
             if (IsReservedDeviceName(path))
             {
-                isDirectory = false;
                 return false;
             }
-
-            int result = NativeMethods.GetFileAttributes(path);
-            if (result == -1)
+#endif
+            try
+            {
+                // Use 'File.GetAttributes()' because we want to get access exceptions.
+                // TODO: we should review the tricky logic
+                // (historically we throw 'UnauthorizedAccessException' here) and migrate
+                // to standard methods 'File.Exist()' and 'Directory.Exists()' where possible.
```

Comment:
Seems my comment is not clear.
I meant that we should migrate from the  ItemExists method to File.Exists() and Directory.Exits() which don't throw UnauthorizedAccessException.

Reply:
Yeah, I agree. I also pointed that out in my comments. Open an issue to track that migration instead of this comment. This comment is not useful in the context of this method, which is designed to throw on error.

Reply:
Done #7031

Reply:
TODO comment was removed.